src/Gateway.php

```diff
@@ -703,6 +743,11 @@ public function check_paytrail_response() {
 		$cancel_order     = filter_input(INPUT_GET, 'cancel_order');
 		$pay_for_order    = filter_input(INPUT_GET, 'pay_for_order');
 
+		// Store information that transaction-specific settlement was used
+		if ( $this->transaction_settlement_enable ) {
+			update_post_meta( $order_id, '_paytrail_ppa_transaction_settlement', true );
```

Comment:
Any actions on the orders now need to be HPOS compliant. See https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book

For example this should be replaced with a $order->update_meta_data

You also need to ensure that $order->save() is called to make sure the data is updated. But this should be called as little as possible for performance reasons.